src/components/ScenarioParameters/components/ScenarioActions/components/LaunchButton/LaunchButton.js

```diff
@@ -37,8 +39,21 @@ export const LaunchButton = () => {
     async (event) => {
       event.stopPropagation();
       if (isDirty || forceUpdate) {
-        await processFilesToUpload();
-        saveAndLaunchScenario(currentScenarioId, getParametersToUpdate());
+        const error = await processFilesToUpload();
+        if (error == null) {
+          saveAndLaunchScenario(currentScenarioId, getParametersToUpdate());
+        } else {
+          applicationStore.dispatch(
+            dispatchSetApplicationErrorMessage(
```

Comment:
We're in a React component, we can use hooks instead of calling `applicationStore.dispatch`.
Can you modify this part to use the `useSetApplicationErrorMessage` hook (an example of this can be found in the scenario view component)